test/allowlist_for_publicAPI.json

```diff
@@ -3418,8 +3418,10 @@
     "Callable",
     "IterDataPipe",
     "Iterator",
+    "Optional",
     "TypeVar",
     "check_lambda_fn",
+    "deprecation_warning",
```

Comment:
Add this two APIs into allowlist for the failing test. cc: @albanD 
Is this the right way to do so? `deprecation_warning` can be changed to `_deprecation_warning` respecting the private method. But, I am curious about why all `typing` related classes are identified as public API?

Reply:
Based on my understanding, I don't think we want users to use the `deprecation_warning`, so we should add `_` in front of it.

Unsure about the `typing` part, @albanD can answer.

Reply:
Agree. But, I am reluctant to change the name in this PR considering we need to change the usage of `deprecation_warning` accordingly in TorchData. I would support to change it in a separate PR.

Reply:
> Agree. But, I am reluctant to change the name in this PR considering we need to change the usage of `deprecation_warning` accordingly in TorchData. I would support to change it in a separate PR.

I see. That makes sense.

Reply:
Hey!
Not the right way no haha
In general, nothing should ever be added to this list.
See details in https://github.com/pytorch/pytorch/wiki/Public-API-definition-and-documentation for what this tests.

To avoid having the `typing` class being identified as public, the simple solution is to add an `__all__` to your module that will specify what is public or not!

Reply:
I see. Then, I will open a new PR to fix them and land the fix before this PR.

Reply:
Sounds like the simplest way yes.
Feel free to add me a as a reviewer there if you need.